### DISCUSSION TITLE
Why can't I have the plot-create-disk() source so I can fix the "Kill a NVME" problem?

### DISCUSSION THREAD

**Discussion by room101-dev** [Chia-Network/chia-blockchain#5373] [General]
Why don't we have a 'pool' the Chinese have had one all along?? Well these questions get deeper than this.

The more important rhetorical question is "IS the entire project", really a chinese operation?? Don't joke, think about bitcoin-gold, it was funded by China, and the dev's were in Ukraine, they down-played the Chinese control; Told everybody is was a "Ukranian Project"

Think about this too, we can't even see the source for the plot-creation in chia, because its top secret, that is VERY chinese not making source public, CHIA likes to say 'the source is on github', NO ITS NOT, not the source that counts.

I'm burning up a $200 NVME every two weeks running this plot-create software, and if I had the source I would fix the problem, just like I do in GPU mining. I don't burn up my GPU cards, and I hate to see my NVME's dying when they're only at 25% of TBW ( warranty ) ( samsung 980 pro 's )

Lot's of things I could do, if I had  the source, to prevent my NVME's from getting fried. But no progress is being made by CHIA-DEV, they just blow off all the complaints. The entire reason we have GITHUB is to share the source, and let problems get fixed in the wild.

IMHO just like the 'pool' China has a pool for CHIA all along, me thinks China is 6+ months ahead of CHIA(github) in every way, I'm going to have to assume the real reason we can't have the source for plot-create-disk(), is that its OWNED by China, the guys who funded the CHIA project. Which explains why "Going public, IPO" is the basis of the entire project, rather than creating a rock-solid POS project as promised.

Well heck, maybe its all owned by Samsung, and they just want to create a market for EVO 980's by having them all die?? Warrant's only cover a small percentage of TBW, and most 'mining' (farming-blah) burns  25% of warranty TBW in 2 weeks, even though the NVME can no longer be used for -t (temp-dir), it can be used for -2 (2nd temp dir) because large contiguous files still work, its only the random access stuff that gets killed in the controller, early death of new NVME's. But it doesn't all matter, cuz in 2 months the warranty's are kaput. recall that 5yr is the promised warranty, but that's only for normal mortal usage.

If the problems of burning up NVME's isn't fixed this project will fail, perhaps nobody cares; If the only goal is an IPO, then to the moon. If the problems continue to be ignored, it means more CHIA-Clones in the wild, and most likely 100% open-source. Chia isn't/wasn't the first POS, but it was the first to 'market the hell out of the idea'; just like a business that intended to go public.

All the marketing like "Use old machines" ( impossible, they only supported 4gb files ), or use old hard-disks ( impossible ) you 'plot' an HDD and its shot the first time you do so, even SSD's can't be used for 'plotting'; So what's green about filling the land-fills with NVME drives?

**Comment by bill-95** [comment]
>I'm burning up a $200 NVME every two weeks running this plot-create software

I think there's something wrong with the NVME you are using...

> its only the random access stuff that gets killed in the controller, early death of new NVME's.

Curious to know where you get this from?

If you want to improve the plotter, the source is right here: https://github.com/Chia-Network/chiapos

**Reply by climba03003** [reply]
He is blind to see the sources code. https://github.com/Chia-Network/chiapos

**Reply by room101-dev** [reply]
> He is blind to see the sources code. https://github.com/Chia-Network/chiapos

Thank You SO much :)
On download from chia-net/blockchain, it doesn't  bring in chiapos, the python wrapper is create_plot_disk(), I search everywhere on the mother site for this code, the actual c++ is plotter_disk.hpp
So now I have have what I need, I even searched 'plotter' and never hit this file.
As soon as you provided the chiapos link I was able to find all the source I was looking for, now I can hack the plotter, and figure out how to keep from killing my NVEM's :)
